### PR TITLE
Test composer cache

### PR DESCRIPTION
**Pull request by Plytas**

### CODE REVIEW

**Comment by Plytas** [comment]
Won't work since `composer.lock` isn't committed to repo.

**Comment by Plytas** [comment]
Alright so this is not ideal, but it works. Cache is stored with a key, that key is made by hashing `composer.json` .

**It works**, laravel 7 and 8 uses different keys because different laravel version gets specified in composer.
**It's not ideal**, because it cached current package versions and the cache is gonna provide them until we update composer.json again. This means that actions will still run fine, but we'll see more and more packages say Downloading instead of Loading from cache.

Still, it's better than downloading them every time.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -67,6 +67,19 @@ jobs:
...
+              run: |
+                echo "::set-output name=dir::$(composer config cache-files-dir)"
+              
+            - name: Cache Composer Downloads
+              uses: actions/cache@v2
+              with:
+                  path: ${{ steps.composer-cache.outputs.dir }}
+                  key: ${{ runner.os }}-composer--${{ hashFiles('**/composer.json') }}
+                  restore-keys: |
+                    ${{ runner.os }}-composer-
```
Comment (ArlonAntonius) [inline .github/workflows/test.yml:81]: Either this needs to be cut off or it would need to include the laravel version? 🤔

**Comment by ArlonAntonius** [comment]
Is there a way we can fix it so it updates the cache once in a while?

File: .github/workflows/test.yml (hunk 1 above)
Comment (Plytas) [reply .github/workflows/test.yml:81]: Took it from an example. Turns out `restore-keys` are optional and it's used as an alternative key, so I'll just remove it.

**Comment by Plytas** [comment]
Ok so this should be good. Cache key consists of commit sha and hash of composer.json. It's also gonna use the restore-key. This means that on new commit it's gonna be a cache miss, but it's gonna find a different cache by partial match using restore-key. So it should load all packages from cache, install any new ones or the ones that got updated and then save new cache since it was technically a cache miss.

File: .github/workflows/test.yml (hunk 1 above)
Comment (Plytas) [reply .github/workflows/test.yml:81]: Removed it and then put it back in 😄 I think we can leave it like this as a default cache key that will always get a hit.
Reference: https://docs.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action

**Pull request merged**